cmake/find/base64.cmake

```diff
@@ -1,4 +1,8 @@
-option (ENABLE_BASE64 "Enable base64" ${ENABLE_LIBRARIES})
+if(NOT ARCH_PPC64LE)
```

Comment:
Ok. I see that "Turbo Base64" has generic code inside and it can be enabled. But the library is not organized in a way to make it easy.

Better to replace `NOT ARCH_PPC64LE` to `ARCH_AMD64`.

Reply:
About base64, it has ARM Neon support too. So maybe `ARCH_AMD64 OR ARCH_ARM` ?
It also seems to have detection for powerpc (but no powerpc specific code), but i guess something is broken there because of the errors above.

Reply:
It may not compile for ARM out of the box (the library has complicated structure), so better not to bother.

Reply:
It does get compiled fine for aarch64 out of the box, i never had to disable it.
https://build.opensuse.org/public/build/home:mastertheknife:clickhouse-lts/CentOS_8/aarch64/ClickHouse/_log
```[ 1213s] [2594/5736] /usr/bin/clang  -I../contrib/sentry-native/include -isystem ../contrib/libcxx/include -isystem ../contrib/libcxxabi/include -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fasynchronous-unwind-tables  -fuse-ld=lld -fdiagnostics-color=always  -gdwarf-aranges -pipe  -Wall -Wno-unused-command-line-argument  -fdiagnostics-absolute-paths -fexperimental-new-pass-manager -Werror -w -O2 -g -DNDEBUG -O3  -flto=thin   -D OS_LINUX -falign-loops -MD -MT contrib/base64-cmake/CMakeFiles/base64_scalar.dir/__/base64/turbob64d.c.o -MF contrib/base64-cmake/CMakeFiles/base64_scalar.dir/__/base64/turbob64d.c.o.d -o contrib/base64-cmake/CMakeFiles/base64_scalar.dir/__/base64/turbob64d.c.o -c ../contrib/base64/turbob64d.c
[ 1213s] [2595/5736] /usr/bin/clang  -I../contrib/sentry-native/include -isystem ../contrib/libcxx/include -isystem ../contrib/libcxxabi/include -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fasynchronous-unwind-tables  -fuse-ld=lld -fdiagnostics-color=always  -gdwarf-aranges -pipe  -Wall -Wno-unused-command-line-argument  -fdiagnostics-absolute-paths -fexperimental-new-pass-manager -Werror -w -O2 -g -DNDEBUG -O3  -flto=thin   -D OS_LINUX -falign-loops -MD -MT contrib/base64-cmake/CMakeFiles/base64_scalar.dir/__/base64/turbob64c.c.o -MF contrib/base64-cmake/CMakeFiles/base64_scalar.dir/__/base64/turbob64c.c.o.d -o contrib/base64-cmake/CMakeFiles/base64_scalar.dir/__/base64/turbob64c.c.o -c ../contrib/base64/turbob64c.c
[ 1213s] [2596/5736] /usr/bin/clang  -I../contrib/sentry-native/include -isystem ../contrib/libcxx/include -isystem ../contrib/libcxxabi/include -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fasynchronous-unwind-tables  -fuse-ld=lld -fdiagnostics-color=always  -gdwarf-aranges -pipe  -Wall -Wno-unused-command-line-argument  -fdiagnostics-absolute-paths -fexperimental-new-pass-manager -Werror -w -O2 -g -DNDEBUG -O3  -flto=thin   -D OS_LINUX -falign-loops -MD -MT contrib/base64-cmake/CMakeFiles/base64_ssse3.dir/__/base64/turbob64sse.c.o -MF contrib/base64-cmake/CMakeFiles/base64_ssse3.dir/__/base64/turbob64sse.c.o.d -o contrib/base64-cmake/CMakeFiles/base64_ssse3.dir/__/base64/turbob64sse.c.o -c ../contrib/base64/turbob64sse.c
[ 1213s] [2598/5736] : && /usr/bin/cmake -E rm -f contrib/base64-cmake/libbase64.a && /usr/bin/llvm-ar qc contrib/base64-cmake/libbase64.a  contrib/base64-cmake/CMakeFiles/base64_scalar.dir/__/base64/turbob64c.c.o contrib/base64-cmake/CMakeFiles/base64_scalar.dir/__/base64/turbob64d.c.o contrib/base64-cmake/CMakeFiles/base64_ssse3.dir/__/base64/turbob64sse.c.o && /usr/bin/llvm-ranlib contrib/base64-cmake/libbase64.a && :
```
I will fix the `NOT ARCH_PPC64LE` to `ARCH_AMD64 OR ARCH_ARM`